util/codec/codec.go

```diff
@@ -157,7 +157,12 @@ func EstimateValueSize(sc *stmtctx.StatementContext, val types.Datum) (int, erro
 	case types.KindString, types.KindBytes:
 		l = valueSizeOfBytes(val.GetBytes())
 	case types.KindMysqlDecimal:
-		l = valueSizeOfDecimal(val.GetMysqlDecimal(), val.Length(), val.Frac()) + 1
+		var err error
+		l, err = valueSizeOfDecimal(val.GetMysqlDecimal(), val.Length(), val.Frac())
```

Comment:
Maybe can change to 'err := ' instead of 'var err error', to be in accordance with other 'err := '

Reply:
This may redefine ``l`` as a local variable in this case, which leads to a wrong logic.

Reply:
Right